### DISCUSSION TITLE
how to only allow trailing slash?

### DISCUSSION THREAD

**Discussion by tsulatsitamim** [nuxt/nuxt#18391] [Questions]
```
  router: {
    trailingSlash: true,
    options: {
      strict: true,
    },
  },
```

use this setting return 404 with trailing slash and return 200 without trailing slash

**Comment by michael-picard** [comment]
I have the same issue, what I just noticed is that on full page reload routes using `[slug]` remove the trailing slash but routes with `[...slug]` keep the trailing slash! This does seem like a bug.

**Comment by marekmis** [comment]
Found the same problem. I can confirm the behaviour is different on full page reload of routes using `[slug]` and the static ones. In my case I discovered the problem due to dealing with query parameters. I found few solutions for this problem, but the best so far is below, no 3rd party dependencies and it's handling the query and hash nicely:

```
// middleware/noTrailingSlash.global.js

export default defineNuxtRouteMiddleware((to) => {
  if (to.path.endsWith('/') && to.path !== '/') {

    const { path, query, hash } = to
    const nextPath = path.replace(/\/+$/, '') || '/'
    const nextRoute = { path: nextPath, query, hash }

    return navigateTo(nextRoute, { redirectCode: 301 })
  }
})

```
